packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/Translator.java

```diff
@@ -63,6 +64,11 @@
     info.put("developerPayload", purchase.getDeveloperPayload());
     info.put("isAcknowledged", purchase.isAcknowledged());
     info.put("purchaseState", purchase.getPurchaseState());
+    AccountIdentifiers accountIdentifiers = purchase.getAccountIdentifiers();
+    if (accountIdentifiers != null) {
```

Comment:
There is already a test checking that the `obfuscatedAccountId` and `obfuscatedProfileId` are present in the `HashMap` when the the `accountIdentifiers` object is not `null`. However I think it is also good to test that the keys are not present when the `accountIdentifiers` object is `null`.

This guards against somebody removing the `if` statement without thinking about it.

Reply:
@mvanbeusekom this is difficult to test, because although getAccountIdentifiers is nullable, I don't think it will ever be null.

This is the decompiled source code of `getAccountIdentifiers()`
```
    @Nullable
    public AccountIdentifiers getAccountIdentifiers() {
        String var1 = this.zzc.optString("obfuscatedAccountId");
        String var3 = this.zzc.optString("obfuscatedProfileId");
        if (var1 == null && var3 == null) {
            return null;
        } else {
            AccountIdentifiers var2 = new AccountIdentifiers(var1, var3);
            return var2;
        }
    }
```

The account and profile id are retrieved from the JSONObject using [optString(key)](https://developer.android.com/reference/org/json/JSONObject?authuser=1#optString(java.lang.String)).

> Returns the value mapped by name if it exists, coercing it if necessary, or the empty string if no such mapping exists.

Not setting the `obfuscatedAccountId` and `obfuscatedProfileId` in the json will lead to getting a `new AccountIdentifiers("", "")`. Setting them to `null` leads to the same result as that is ignored by the JSONObject decoder. It is marked as `@Nullable` so we can't remove `if (accountIdentifiers != null)`, but it is actually never `null` in the current implementation.

Reply:
What if you pass in a mock instance of the `Purchase` class? Something like:

```java
class FakePurchase extends Purchase {
  @Override
  public AccountIdentifiers getAccountIdentifiers() {
    return null;
  }
}
```